docs/test/test-guide.adoc

```diff
@@ -0,0 +1,30 @@
+= Test Guide
+:toc: macro
+:toc-title:
+:toclevels: 1
+
+toc::[]
+
+== Setting up test environment
+
+Requires *Go 1.14*
+
+Testing happens with the above version. Developers are advised to stick to this version.
```

Comment:
We should definitely clarify, as most developers are probably on the latest supported version of Go.

Reply:
I will add more clarification in the doc